### PR TITLE
Provide quickfixes for illegal rule references (WIP)

### PR DESCRIPTION
**Pull request by nbhusare**
- Enhanced the quick-fix to handle case where a terminal-fragment is referenced as a hidden token in a Parser rule definition
- Added method `createChangeToIssueResolution()` to avoid duplicate 'Change to' quick-fix if there are multiple/different fixes for an issue
- Added quick-fix method `fixUnresolvedEnumRule()`.

Signed-off-by: nbhusare <neerajbhusare@gmail.com>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java [hunk 1]
```diff
@@ -146,32 +148,53 @@ public void fixUnresolvedRule(final Issue issue, IssueResolutionAcceptor accepto
...
+		String ruleType = "terminal ";
 		acceptor.accept(issue, //
 				"Create terminal '" + ruleName + "'", // 
 				"Create terminal '" + ruleName + "'", //
 				NULL_QUICKFIX_IMAGE, //
-				createNewRule(ruleName, false));
-		createLinkingIssueResolutions(issue, acceptor);
+				createNewRule(ruleName, ruleType));
+
+		Optional<IssueResolution> iResolution = acceptor.getIssueResolutions().stream().filter(ir -> {
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java:166]: i am not sure if i like the after effect filtering
shouldnt 
createLinkingIssueResolutions
be modified to ctreate resolutions for that matching methods rule type only

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java:166]: @szarnekow what do you think?

Comment (szarnekow) [reply org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java:166]: I'd be in favor of adding another `@Fix(UNRESOLVED_TERMINAL_RULE)` and call `createLinkingIssueResolutions()` from there to make sure it is only invoked once for that issue code.

**Comment by nbhusare** [comment]
@szarnekow The code changes are good to be reviewed. I haven't fixed the unit-tests which is why I have put WIP.

**Comment by cdietrich** [comment]
@nbhusare the 19 commits look wired? did you properly rebase on eclipse/master?

**Comment by nbhusare** [comment]
@nbhusare I had rebased correctly. In Eclipse, I saw that my branch was lagging behind some comments in remote after which I pulled which lead to this state.

**Comment by cdietrich** [comment]
Bildschirmfoto 2020-08-10 um 11 40 43
there is this merge commit.
on rebases there should be no merges

**Comment by nbhusare** [comment]
@cdietrich I am not sure. I don't merge except on the master (git fetch upstream, git merge upstrem/master, git push) after which I switch to the feature branch and do a git rebase master.

**Comment by nbhusare** [comment]
@cdietrich Probably the git pull on the feature branch might have created the merge commit. I am not sure.

**Comment by nbhusare** [comment]
@cdietrich I'll be discarding the branch after porting the changes to a new one.

**Comment by cdietrich** [comment]
can you please try to clean that up

- undo last commit (e.g. git reset HEAD~)
- rebase on eclipse/master

or

- move create a new branch from yours
- on the pr branch, reset to origin master and cherry-pick your commit

or create a new branch?

**Comment by nbhusare** [comment]
@cdietrich I have created a new branch - https://github.com/eclipse/xtext-eclipse/pull/1507
@szarnekow Please use pull/1507 for any further review.

**Pull request closed without merging**